webservices/resources/legal.py

```diff
@@ -85,18 +85,29 @@ def get(self, q='', from_hit=0, hits_returned=20, **kwargs):
         else:
             doc_types = [kwargs.get('type')]
 
+            # if doc_types is not in one of these 4 doc types :'statutes', 'regulations', 'advisory_opinions', 'murs'
+            # then reset type = all (= four default search types)
+            if doc_types[0] not in ('statutes', 'regulations', 'advisory_opinions', 'murs'):
+                doc_types = ['statutes', 'regulations', 'advisory_opinions', 'murs']
```

Comment:
This is a great start in accounting for this and resolves the issue! :-)  I think there are a couple of opportunities for improvement, too:

* I noticed that you're checking for the presence of the value provided in a `tuple` instead of a `list` (you have a collection of values wrapped in parentheses instead of square brackets).  This isn't inherently bad, especially in this case as you're just checking to see if the value exists from a known set, but it might lead someone to infer that the value you're checking (`doc_types[0]`) is also in a `tuple`, which it is not (it's a `list` as defined above on line 86).
* This collection of values are being repeated on three lines that I see, so this might be an opportunity for a bit of refactoring to clean this up a bit and prevent duplication (also known as DRY - Don't Repeat Yourself).

In terms of refactoring, I'd approach it like this:

* Create a new constant, defined just under the `INNER_HITS` constant you see at lines 18 - 27 in the same file), and call it `ALL_DOCUMENT_TYPES` or something to that effect, and set it like this:  `ALL_DOCUMENT_TYPES = ['statutes', 'regulations', 'advisory_opinions', 'murs',]`
* Change line 84 to be this instead:  `doc_types = ALL_DOCUMENT_TYPES`
* Change line 90 to this:  `if doc_types[0] not in ALL_DOCUMENT_TYPES`
* Change line 91 to this:  `doc_types = ALL_DOCUMENT_TYPES`

Does this make sense?  By doing this, you're only defining the document types in one place so if they ever change in the future, it's just the constant that has to be modified and all of the rest of the logic leverages the constant.  Let me know if you have any questions or other ideas!

Reply:
great suggestion. every clean and clear. will do.
Thanks.